File: python/grid.py
class Grid:
    def __init__(self, width, height):
        self.width = width
        self.height = height
        self.cells = [[False] * height for _ in range(width)]

    def set_alive(self, x, y, alive):
        x = x % self.width
        y = y % self.height
        self.cells[x][y] = alive

    def get_alive(self, x, y):
        x = x % self.width
        y = y % self.height
        return self.cells[x][y]

    def alive_neighbours(self, x, y):
        result = 0
        for i in range(x - 1, x + 2):
            for j in range(y - 1, y + 2):
                if (i != x or j != y) and self.get_alive(i, j):
                    result += 1
        return result

    def step_all(self, next):
        self.step(next, 0, self.width, 0, self.height)

    def step(self, next, startX, endX, startY, endY):
        for x in range(startX, endX):
            for y in range(startY, endY):
                a = self.alive_neighbours(x, y)
                if a == 3:
                    next.set_alive(x, y, True)
                elif a == 2:
                    next.set_alive(x, y, self.get_alive(x, y))
                else:
                    next.set_alive(x, y, False)

    def __str__(self):
        return '\n'.join([
            ''.join([
                'X ' if self.get_alive(x, y) else '. '
                for x in range(self.width)
            ])
            for y in range(self.height)
        ])

File: java/Grid.java
public class Grid {
    public final int width;
    public final int height;
    private final boolean[][] cells;

    public Grid(int width, int height) {
        this.width = width;
        this.height = height;
        this.cells = new boolean[width][height];
    }

    void setAlive(int x, int y, boolean alive) {
        x = x % width;
        y = y % height;
        if (x < 0) x += width;
        if (y < 0) y += height;
        cells[x][y] = alive;
    }

    boolean getAlive(int x, int y) {
        x = x % width;
        y = y % height;
        if (x < 0) x += width;
        if (y < 0) y += height;
        return cells[x][y];
    }

    int aliveNeighbours(int x, int y) {
        int result = 0;
        for (int i = x - 1; i < x + 2; i++) {
            for (int j = y - 1; j < y + 2; j++) {
                if ((i != x || j != y) && getAlive(i, j)) {
                    result += 1;
                }
            }
        }
        return result;
    }

    void stepAll(Grid next) {
        step(next, 0, width, 0, height);
    }

    void step(Grid next, int startX, int endX, int startY, int endY) {
        for (int x = startX; x < endX; x++) {
            for (int y = startY; y < endY; y++) {
                int a = aliveNeighbours(x, y);
                if (a == 3) {
                    next.setAlive(x, y, true);
                } else if (a == 2) {
                    next.setAlive(x, y, getAlive(x, y));
                } else {
                    next.setAlive(x, y, false);
                }
            }
        }
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                if (getAlive(x, y)) {
                    sb.append('X');
                } else {
                    sb.append('.');
                }
                sb.append(' ');
            }
            sb.append('\n');
        }
        return sb.toString();
    }
}
